airflow/podman.go

```diff
@@ -0,0 +1,236 @@
+	podmanCmd := Command{
+		Command: "podman",
+		Args:    []string{"machine", "stop", name},
+	}
+	output, err := podmanCmd.Execute(" stopping podman machine...")
+	if err != nil {
+		return fmt.Errorf("error stopping Podman machine: %s, output: %s", err, output)
+	}
+	return nil
+}
+
+func deletePodmanMachine(name string) error {
+	podmanCmd := Command{
+		Command: "podman",
+		Args:    []string{"machine", "rm", "-f", name},
+	}
+	output, err := podmanCmd.Execute(" removing podman machine...")
+	if err != nil {
+		return fmt.Errorf("error removing Podman machine: %s, output: %s", err, output)
+	}
+	return nil
+}
+
+// InspectPodmanMachine inspects a given podman machine name.
+func InspectPodmanMachine(machineName string) (*Machine, error) {
+	podmanCmd := Command{
+		Command: "podman",
+		Args:    []string{"machine", "inspect", machineName},
+	}
+	output, err := podmanCmd.Execute("")
+	if err != nil {
+		return nil, fmt.Errorf("error inspecting Podman machine: %s, output: %s", err, output)
+	}
+
+	var machines []Machine
+	err = json.Unmarshal([]byte(output), &machines)
+	if err != nil {
+		return nil, err
+	}
+	if len(machines) == 0 {
+		return nil, fmt.Errorf("machine not found: %s", machineName)
+	}
+
+	return &machines[0], nil
+}
+
+// ListPodmanMachines lists all Podman machines.
+func ListPodmanMachines() ([]Machine, error) {
+	podmanCmd := Command{
+		Command: "podman",
+		Args:    []string{"machine", "ls", "--format", "json"},
+	}
+	output, err := podmanCmd.Execute("Listing podman machine...")
+	if err != nil {
+		return nil, fmt.Errorf("error listing Podman machines: %s, output: %s", err, output)
+	}
+	var machines []Machine
+	err = json.Unmarshal([]byte(output), &machines)
+	if err != nil {
+		return nil, err
+	}
+	return machines, nil
+}
+
+func FindMachineByName(items []Machine, name string) *Machine {
+	for _, item := range items {
+		if item.Name == name {
+			return &item // Return a pointer to the found item
+		}
+	}
+	return nil // Return nil if no item was found
+}
+
+func InitPodmanMachineCMD() error {
+	projectName, err := ProjectNameUnique()
+	if err != nil {
+		return fmt.Errorf("error retrieving working directory: %w", err)
+	}
+	machineName := "astro-" + formatMachineName(projectName)
+
+	machine, err := InspectPodmanMachine(machineName)
+	if machine != nil {
+		if machine.State == "stopped" {
+			err = StartPodmanMachine(machineName)
+			if err != nil {
+				return fmt.Errorf("error starting Podman machine: %s", err)
+			}
+		}
+		setDockerHost(machine)
+		return nil
+	}
+
+	podmanCmd := Command{
+		Command: "podman",
+		Args:    []string{"machine", "init", machineName, "--now"},
+	}
+	output, err := podmanCmd.Execute(" initializing podman machine")
+	if err != nil {
+		return fmt.Errorf("error starting Podman machine: %s, output: %s", err, output)
+	}
+
+	machine, err = InspectPodmanMachine(machineName)
+	if err != nil {
+		return err
+	}
+	setDockerHost(machine)
+	return nil
+}
+
+func SetPodmanDockerHost() error {
```

Comment:
@pritt20 when we start the machine, I think we also would want to run `podman system connection default $MACHINE_NAME` to ensure that if there's another machine or the default, default machine that we are targeting the one for this project. What do you think?

Reply:
Yes, I am already testing this. I'll push the code once I complete the changes.